cdap-app-fabric/src/main/java/io/cdap/cdap/internal/app/deploy/pipeline/DeleteAndCreateSchedulesStage.java

```diff
@@ -43,6 +43,10 @@ public DeleteAndCreateSchedulesStage(Scheduler programScheduler) {
 
   @Override
   public void process(final ApplicationWithPrograms input) throws Exception {
+    // if a workflow doesn't exist, the remove the schedules
+    if (input.getSpecification().getWorkflows().isEmpty()) {
+      programScheduler.deleteSchedules(input.getApplicationId());
+    }
```

Comment:
@yaojiefeng @albertshau fixing https://cdap.atlassian.net/browse/CDAP-20403 here

Reply:
I don't think this will be a complete solution. For example, an application can have 2 workflows, but only delete one. The deleted one has schedules for it and those should be removed.

That being said, it looks like this entire class is meant to delete the schedules, with the logic in the below loop meant for deleting them. It should be something like the deleteSchedule() call not working properly or the listSchedules() call not returning the actual schedules.

Reply:
It would also be better to do this in a separate PR and commit, as it's not related to the LCM flag.

Reply:
so would it make sense to check for the corresponding workflow for a schedule- if it doesn't exist, then delete it in the loop below?

Reply:
That's already what the loop below is trying to do, but for whatever reason it is not working anymore. Need to understand why and do the proper fix

Reply:
so because of the check at L51 - we return early and do not delete the schedules in the for loop below.  In the test - both `canUpdateSchedules` flag and `workflow` flags are set to false.